### ISSUE TITLE
bug: cannot use `Attach to Session` when using LambdaTest cloud provider

### ISSUE THREAD

**Issue by Delta456** [appium/appium-inspector#1652]
### Is this an issue specifically with Appium Inspector?

- [X] I have verified that my issue does _not_ occur with Appium, and should be investigated as an Appium Inspector issue

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When trying to use Appium inspector's `Attach to Session` using the LambdaTest cloud provider and entering all credentials. The session ID also automatically comes but when clicked on one it gives `Cannot read property 'match' of undefined` as shown.

<img width="1212" alt="image" src="https://github.com/user-attachments/assets/0250d770-8c40-4781-a800-0b68b03285f6">

### Expected Behavior

I expected `Attach to Session` to work for LambdaTest as other Cloud Providers seem to work correctly

### Operating System

Mac

### Appium Inspector Version

2024.8.2

### Further Information

Perhaps this would be important

<img width="1212" alt="image" src="https://github.com/user-attachments/assets/259a4e3e-0a90-4c97-bb1f-acca29bdc6d9">

Also `Attach to Session` used to work in **versions <= `2023.3.1` release**.

**Comment by eglitise** [comment]
This seems like a LambdaTest-specific issue, since they use a slightly different response format. I believe a fix was attempted in #1369, but it spun off into a separate discussion. Looking at the proposed change again, and knowing the adjustments made to have the LambdaTest session show up under Attach to Sessions, I think that fixing this will be fairly straightforward.

**Comment by eglitise** [comment]
The change should be made in `app/common/renderer/actions/Session.js`, lines `553-555`:
```js
        if (attachedSession) {
          attachedSessionCaps = attachedSession.capabilities;
        } else {
```
I think this could work as a fix:
```js
        if (attachedSession) {
          attachedSessionCaps = attachedSession.capabilities;
          if (session.serverType === SERVER_TYPES.LAMBDATEST) {
            // adjust for LambdaTest-specific format
            if ('capabilities' in attachedSessionCaps) {
              attachedSessionCaps = attachedSessionCaps.capabilities;
            }
            if ('desired' in attachedSessionCaps) {
              attachedSessionCaps = attachedSessionCaps.desired;
            }
          }
        } else {
```
A PR would be greatly appreciated :) especially because I can't test this myself (I only have local devices on hand)
